include/mbedtls/ssl.h

```diff
@@ -1374,6 +1378,46 @@ void mbedtls_ssl_set_bio( mbedtls_ssl_context *ssl,
                           mbedtls_ssl_recv_t *f_recv,
                           mbedtls_ssl_recv_timeout_t *f_recv_timeout );
 
+#if defined(MBEDTLS_SSL_PROTO_DTLS)
+/**
+ * \brief          Set the Maximum Tranport Unit (MTU).
+ *                 Special value: 0 means unset (no limit).
+ *                 This represents the maximum size of a datagram payload
+ *                 handled by the transport layer (usually UDP) as determined
+ *                 by the network link and stack. In practice, this controls
+ *                 the maximum size datagram the DTLS layer will pass to the
+ *                 \c f_send() callback set using \c mbedtls_ssl_set_bio().
+ *
+ * \note           This can be called at any point during the connection, for
+ *                 example when a PMTU estimate becomes available from other
+ *                 sources, such as lower (or higher) protocol layers.
+ *
+ * \note           This only controls the size of the packets we send.
+ *                 Client-side, you can request the server to use smaller
+ *                 records with \c mbedtls_ssl_conf_max_frag_len().
+ *
+ * \note           If both a MTU and a maximum fragment length have been
+ *                 configured (or negotiated with the peer), the resulting
+ *                 lower limit (after translating the MTU setting to a limit
+ *                 on the record content length) is used.
+ *
+ * \note           This can only be used to decrease the maximum size
+ *                 of datagrams sent. It cannot be used to increase the
+ *                 maximum size of records over the limit set by
+ *                 #MBEDTLS_SSL_OUT_CONTENT_LEN.
```

Comment:
I think I disagree here. `MBEDTLS_SSL_OUT_BUFFER_LEN` is an internal macro (defined in `ssl_internal.h`), so I'm reluctant to mention it in user-facing documentation. I took care to mention "size of the records" rather than "of the datagrams" in the beginning of the sentence for consistency with the macro I'm mentioning. I'm now adding something to the previous sentence to make the connection between datagram size and record size more apparent, but I really think the only macro we should mention here is `MBEDTLS_SSL_OUT_CONTENT_LEN`.

Reply:
Sorry for being picky here, but what I'm missing is the distinction between plain and ciphertext. What about saying `This can only be used to decrease the maximum size of datagrams (hence protected records, as those cannot span multiple datagrams). It cannot be used to increase the maximum size of protected records over the limit implied by the maximum record payload size #MBEDTLS_SSL_OUT_CONTENT_LEN`.?

Reply:
I think what you are suggesting is correct, but heavier that necessary. (Also, the text in parenthesis could be slightly misleading: unprotected records can't span multiple datagrams either.) The MTU is a limit on the size of protected records, which is practice is then converted to a limit on the size of unprotected records (this is already hinted at in the previous note). By switching to that view, the second part of the sentence is slightly more straightforward.

Reply:
OK, I just pushed a new version trying to clarify this further.